### PR TITLE
Update RobotBase for 2018

### PR DESCRIPTION
**Pull request by auscompgeek**
Also change this to use our NetworkTables global, which points to the
default NetworkTablesInstance.

As a bonus, also use type hints, to aid static analysis tools that don't
understand Sphinx type field directives.

### CODE REVIEW

**Comment by virtuald** [comment]
It would be cool if we could run some of those tools as part of the tests.

File: wpilib/wpilib/robotbase.py [hunk 1]
```diff
@@ -37,98 +36,85 @@ class RobotBase:
...
         """Free the resources for a RobotBase class."""
-        # TODO: delete?
         pass
 
     @staticmethod
-    def isSimulation():
-        """
-            :returns: If the robot is running in simulation.
-            :rtype: bool
+    def isSimulation() -> bool:
```
Comment (Twinters007) [inline wpilib/wpilib/robotbase.py]: are we gonna start adding type hinting to stuff?

File: wpilib/wpilib/robotbase.py [hunk 2]
```diff
@@ -37,98 +36,85 @@ class RobotBase:
...
 
     @staticmethod
-    def isReal():
-        """
-            :returns: If the robot is running in the real world.
-            :rtype: bool
+    def isReal() -> bool:
+        """Get if the robot is real.
+
+        :returns: If the robot is running in the real world.
```
Comment (virtuald) [inline wpilib/wpilib/robotbase.py]: ... returns True?

File: wpilib/wpilib/robotbase.py [hunk 3]
```diff
@@ -160,9 +146,9 @@ def main(robot_cls):
             robot = robot_cls()
         except:
             from .driverstation import DriverStation
-            DriverStation.reportError("ERROR Could not instantiate robot", True)
-            logger.exception("Robots don't quit!")
-            logger.exception("Could not instantiate robot "+robot_cls.__name__+"!")
+            DriverStation.reportError("Unhandled exception instantiating robot " + robot_cls.__name__, True)
+            DriverStation.reportWarning("Robots should not quit, but yours did!")
```
Comment (Twinters007) [inline wpilib/wpilib/robotbase.py]: Took them long enough to change this 🙏

File: wpilib/wpilib/robotbase.py (hunk 1 above)
Comment (virtuald) [reply wpilib/wpilib/robotbase.py]: There is an issue that states we should do it. Adding it gradually is fine.

File: wpilib/wpilib/robotbase.py (hunk 2 above)
Comment (Twinters007) [reply wpilib/wpilib/robotbase.py]: ?

Comment (auscompgeek) [reply wpilib/wpilib/robotbase.py]: I'm just mirroring what upstream have. ¯\\\_(ツ)_/¯

File: wpilib/wpilib/robotbase.py (hunk 1 above)
Comment (auscompgeek) [reply wpilib/wpilib/robotbase.py]: Per #157, yes.

Whether we'll depend on the `typing` module whilst we still support Python 3.4 is an open question however.  But that's a discussion for over there.

Comment (Twinters007) [reply wpilib/wpilib/robotbase.py]: What if we made a rash decision to not support 3.4 in 2018?

File: wpilib/wpilib/robotbase.py (hunk 2 above)
Comment (Twinters007) [reply wpilib/wpilib/robotbase.py]: Why would it only return true?

**Comment by auscompgeek** [comment]
> It would be cool if we could run some of those tools as part of the tests.

Aside from mypy, all the tools I'm thinking of are pretty much IDEs and IDE tooling.

**Pull request merged**